### PR TITLE
layers: Add validation changes for VK_EXT_extended_dynamic_state2

### PR DESCRIPTION
**Pull request by vkushwaha-nv**

### CODE REVIEW

**Comment by vkushwaha-nv** [comment]
@mark-lunarg @Tony-LunarG @ncesario-lunarg  please review.

File: tests/vklayertests_others.cpp [hunk 1]
```diff
@@ -11921,3 +11921,443 @@ TEST_F(VkLayerTest, MixedTimelineAndBinarySemaphores) {
...
+    dyn_state_ci.dynamicStateCount = size(dyn_states);
+    dyn_state_ci.pDynamicStates = dyn_states;
+    pipe.dyn_state_ci_ = dyn_state_ci;
+    pipe.InitState();
+    m_errorMonitor->SetDesiredFailureMsg(kErrorBit, "VUID-VkGraphicsPipelineCreateInfo-pDynamicStates-04868");
+    pipe.CreateGraphicsPipeline();
+    m_errorMonitor->VerifyFound();
+
+    VkCommandBufferObj commandBuffer(m_device, m_commandPool);
+    commandBuffer.begin();
```
Comment (ncesario-lunarg) [inline tests/vklayertests_others.cpp:11982]: I think you can use [m_commandBuffer](https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/f39a8dc664c64e8558fcef0456861addab1626fd/tests/vkrenderframework.h#L298) if you want to here, but not necessary.

File: tests/vklayertests_others.cpp [hunk 2]
```diff
@@ -11921,3 +11921,443 @@ TEST_F(VkLayerTest, MixedTimelineAndBinarySemaphores) {
...
+        (PFN_vkCmdSetPrimitiveRestartEnableEXT)vk::GetDeviceProcAddr(m_device->device(), "vkCmdSetPrimitiveRestartEnableEXT");
+
+    ASSERT_NO_FATAL_FAILURE(InitRenderTarget());
+
+    CreatePipelineHelper pipe(*this);
+    pipe.InitInfo();
+    const VkDynamicState dyn_states[] = {VK_DYNAMIC_STATE_RASTERIZER_DISCARD_ENABLE_EXT, VK_DYNAMIC_STATE_DEPTH_BIAS_ENABLE_EXT,
+                                         VK_DYNAMIC_STATE_PRIMITIVE_RESTART_ENABLE_EXT};
+    VkPipelineDynamicStateCreateInfo dyn_state_ci = {};
+    dyn_state_ci.sType = VK_STRUCTURE_TYPE_PIPELINE_DYNAMIC_STATE_CREATE_INFO;
```
Comment (ncesario-lunarg) [inline tests/vklayertests_others.cpp:11972]: You should be able to use `LvlInitStruct<VkPipelineDynamicStateCreateInfo>()` now that the updated headers have merged. Not necessary, but will ensure `pNext == null`.

**Review by ncesario-lunarg — APPROVED** [review]
Couple non-essential nits, but looks great!

**Comment by ncesario-lunarg** [comment]
Running through [internal CI](http://erusea:4040/8998).

**Comment by ncesario-lunarg** [comment]
@vkushwaha-nv if you could please separate the tests into a separate commit, I think this good to go unless @Tony-LunarG gets a chance to review and has any comments. Thanks!

**Comment by vkushwaha-nv** [comment]
Sure, I have split them into 2 commits. Or did you mean 2 different pull requests?

**Comment by ncesario-lunarg** [comment]
Nope, 2 separate commits is perfect. Thanks!

**Review by Tony-LunarG — APPROVED** [review]
LGTM!

**Pull request merged**